codec-http/src/main/java/io/netty/handler/codec/http/HttpHeaderUtil.java

```diff
@@ -230,7 +231,9 @@ public static void setTransferEncodingChunked(HttpMessage m, boolean chunked) {
             m.headers().add(HttpHeaderNames.TRANSFER_ENCODING, HttpHeaderValues.CHUNKED);
             m.headers().remove(HttpHeaderNames.CONTENT_LENGTH);
         } else {
-            List<CharSequence> values = m.headers().getAll(HttpHeaderNames.TRANSFER_ENCODING);
+            // Make a copy to be able to modify values while iterating
+            List<CharSequence> values =
+                    new ArrayList<CharSequence>(m.headers().getAll(HttpHeaderNames.TRANSFER_ENCODING));
```

Comment:
Good catch. I don't need to copy here any more. @buchgr's PR returned unmodifiable lists, and we no longer do that...so no need to make 2 copies.